src/main/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceV2ApiService.scala

```diff
@@ -0,0 +1,38 @@
+package org.broadinstitute.dsde.firecloud.webservice
+
+import akka.http.scaladsl.model.HttpMethods
+import akka.http.scaladsl.server.Route
+
+import org.broadinstitute.dsde.firecloud.FireCloudConfig
+import org.broadinstitute.dsde.firecloud.service.{FireCloudDirectives, FireCloudRequestBuilding}
+import org.broadinstitute.dsde.firecloud.utils.StandardUserInfoDirectives
+
+import scala.concurrent.ExecutionContext
+
+trait WorkspaceV2ApiService extends FireCloudRequestBuilding with FireCloudDirectives with StandardUserInfoDirectives {
+
+  implicit val executionContext: ExecutionContext
+
+  lazy val rawlsWorkspacesV2Root: String = FireCloudConfig.Rawls.workspacesV2Url
+
+  val workspaceV2Routes: Route =
+    pathPrefix("api" / "workspaces" / "v2") {
+      pathPrefix(Segment / Segment) {
+        (workspaceNamespace, workspaceName) => {
+          val workspaceV2Path = encodeUri(rawlsWorkspacesV2Root + "/%s/%s".format(workspaceNamespace, workspaceName))
+          path("settings") {
+            get {
+              requireUserInfo() { _ =>
+                passthrough(workspaceV2Path + "/settings", HttpMethods.GET)
```

Comment:
consider using `streamingPassthrough` instead (it's generally more performant) - this would also allow you to just proxy the "settings" url for all methods instead of explicitly specifying GET and PUT.